psydac/linalg/identity.py

```diff
@@ -26,9 +26,32 @@ def codomain( self ):
         return self._V
 
     def dot( self, v, out=None ):
+        """
+        Returns the input vector. If out is None or v is the same vector object as out (`v is out`), v is returned (no copy).
+        In all other cases, v is copied to out, and out is returned.
+
+        Parameters
+        ----------
+        v : Vector
+            The vector to return.
+        
+        out : Vector | None
+            Output vector. Has to be either none, or a vector from the same space as v. Behavior is described above.
+        
+        Returns
+        -------
+        Described above.
+        """
         assert isinstance( v, Vector )
         assert v.space is self.domain
-        # find a way to handle out not None
+
+        if out is not None and out is not v:
+            assert isinstance(out, Vector)
+            assert v.space is out.space
+            out *= 0.0
+            out += v
```

Comment:
Is this the place where you would like to use a `copyto` method? Since the `Vector` base class already has a `copy` method, maybe we could just add the optional argument `out` to it. What do you think?

Reply:
Yes, that sounds great! (and yes, this is the place where I would use it) Shall I add it to this PR already? Or to a separate one?

Reply:
Please just make a separate small PR. I'm going to merge this one as soon as you update its description!